core/src/main/scala/scalapb/zio_grpc/ClientMethods.scala

```diff
@@ -1,17 +1,24 @@
 package scalapb.zio_grpc
 
 import io.grpc.CallOptions
+import io.grpc.Deadline
 import java.util.concurrent.TimeUnit
 import zio.Duration
 import zio.UIO
-import io.grpc.Deadline
+import zio.ZIO
 
-trait ClientMethods[Repr] {
+trait ClientMethods[Repr] extends TransformableClient[Repr] {
   // Returns new instance with modified call options
-  def mapCallOptions(f: CallOptions => CallOptions): Repr
+  def mapCallOptions(f: CallOptions => CallOptions): Repr =
+    transform(ZTransform[CallContext, CallContext] { context =>
+      ZIO.succeed(context.copy(options = f(context.options)))
+    })
 
   // Returns new instance with modified metadata
-  def mapMetadataZIO(f: SafeMetadata => UIO[SafeMetadata]): Repr
+  def mapMetadataZIO(f: SafeMetadata => UIO[SafeMetadata]): Repr =
+    transform(ZTransform[CallContext, CallContext] { context =>
+      ZIO.succeed(context.copy(metadata = context.metadata.flatMap(f)))
```

Comment:
I think this may be dangerous. How about:
```
f(context.metadata).map(m => context.copy(metadata = ZIO.succeed(m)))
```
?

Reply:
Interesting, I am curious what danger you had mind - was trying to get my head around it for a few minutes... I think @regiskuckaertz proposal makes sense though since it lifts up the evaluation of `f` so less chance of a code path that causes multiple evaluations of this effect.

Reply:
I changed `ClientCallContext` to contain a `SafeMetadata` instead of `UIO[SaleMetadata]` (this was an artifact of my earlier experimentations) so now `mapMetadataZIO is simply:

```scala
ZTransform[ClientCallContext, ClientCallContext] { context =>
  f(context.metadata).map(metadata => context.copy(metadata = metadata))
}
```